File: packages/java/parser-jvm-plugin-nonnull-kotlin/src/test/kotlin/com/vaadin/hilla/parser/plugins/nonnull/kotlin/basic/data/Person.kt
package com.vaadin.hilla.parser.plugins.nonnull.kotlin.basic.data

import jakarta.persistence.Entity
import jakarta.persistence.OneToMany
import jakarta.persistence.Transient
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import java.time.LocalDate

@Entity
open class Person(
    var firstName: @NotBlank String = "",
    var lastName: String = "",
    var email: @Email String? = null,
    var phone: String = "",
    @OneToMany
    var address: MutableList<Address> = mutableListOf(),
    @Transient
    var team: Map<String, List<Employee?>> = mapOf(),
    var dateOfBirth: LocalDate? = null,
    var occupation: String? = null,
    var role: String? = null,
    var important: Boolean = false,
): AbstractEntity()

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/nonnull/kotlin/basic/snapshots/data/Person.ts
import type AbstractEntity from "./AbstractEntity.js";
import type Address from "./Address.js";
import type Employee from "./Employee.js";
interface Person extends AbstractEntity {
    firstName: string;
    lastName: string;
    email?: string;
    phone: string;
    address: Array<Address>;
    team: Record<string, Array<Employee | undefined>>;
    dateOfBirth?: string;
    occupation?: string;
    role?: string;
    important: boolean;
}
export default Person;
